src/splatting/Splatting.js

```diff
@@ -885,6 +889,34 @@ async function main(initialFilePath) {
     window.addEventListener("resize", resize);
     resize();
 
+    const clamp = (x, low, high) => {
+        return Math.min(Math.max(x, low), high);
+    }
+
+    const remap01 = (x, low, high) => {
+        return clamp((x - low) / (high - low), 0, 1);
+    }
+
+    const _remap01Curve1 = (x, low, high) => {
+        let r = remap01(x, low, high);
+        let a = 2;
+        return (1 - r) * Math.pow(r, a) + r * (-Math.pow(r - 1, a) + 1);
+    }
+
+    const remap01Curve2 = (x, low, high) => {
+        let r = remap01(x, low, high);
+        let a = 2;
+        return -Math.pow(r - 1, a) + 1;
+    }
+
+    const _remap = (x, lowIn, highIn, lowOut, highOut) => {
+        return lowOut + (highOut - lowOut) * remap01(x, lowIn, highIn);
+    }
+
+    const _remapCurve = (x, lowIn, highIn, lowOut, highOut) => {
+        return lowOut + (highOut - lowOut) * remap01Curve2(x, lowIn, highIn);
+    }
```

Comment:
Would it be possible to refactor these common math functions into some kind of utils file, and import them where needed? I notice that similar/same functions to these are also declared in spatialCursor, spatialArrow, and spatialIndicator, which makes me think they could be refactored to a common location

Reply:
Good idea. Will create a math.js and refactor all repeated functions into it.